src/lib/utils/thorchain/memo.ts

```diff
@@ -2,43 +2,60 @@ import { bn } from 'lib/bignumber/bignumber'
 
 import { THORCHAIN_AFFILIATE_NAME } from './constants'
 
-const assertMemoHasPool = (pool: string, memo: string) => {
+const assertMemoHasPool = (pool: string | undefined, memo: string): pool is string => {
   if (!pool) throw new Error(`pool is required in memo: ${memo}`)
+  return true
 }
 
-const assertMemoHasAsset = (asset: string, memo: string) => {
+const assertMemoHasAsset = (asset: string | undefined, memo: string): asset is string => {
   if (!asset) throw new Error(`asset is required in memo: ${memo}`)
+  return true
 }
 
-const assertMemoHasDestAddr = (destAddr: string, memo: string) => {
+const assertMemoHasDestAddr = (destAddr: string | undefined, memo: string): destAddr is string => {
   if (!destAddr) throw new Error(`destination address is required in memo: ${memo}`)
+  return true
 }
 
-const assertMemoHasPairedAddr = (pairedAddr: string, memo: string) => {
+const assertMemoHasPairedAddr = (
+  pairedAddr: string | undefined,
+  memo: string,
+): pairedAddr is string => {
   if (!pairedAddr) throw new Error(`paired address is required in memo: ${memo}`)
+  return true
 }
 
-const assertMemoHasLimit = (limit: string, memo: string) => {
+const assertMemoHasLimit = (limit: string | undefined, memo: string): limit is string => {
   if (!limit) throw new Error(`limit is required in memo: ${memo}`)
+  return true
 }
 
-const assertMemoHasBasisPoints = (basisPoints: string, memo: string) => {
+const assertMemoHasBasisPoints = (
+  basisPoints: string | undefined,
+  memo: string,
+): basisPoints is string => {
   if (!basisPoints) throw new Error(`basis points is required in memo: ${memo}`)
+  return true
+}
+
+const assertMemoHasAction = (action: string | undefined, memo: string): action is string => {
+  if (!action) throw new Error(`action is required in memo: ${memo}`)
+  return true
 }
 
 // Disabling until we validate further, as :MINOUT is optional in the quote response
 // const assertMemoHasMinOut = (minOut: string, memo: string) => {
 //   if (!minOut) throw new Error(`minOut is required in memo: ${memo}`)
 // }
 
-const assertIsValidLimit = (limit: string, memo: string) => {
-  assertMemoHasLimit(limit, memo)
+const assertIsValidLimit = (limit: string | undefined, memo: string) => {
+  if (!assertMemoHasLimit(limit, memo)) return
```

Comment:
If we used the type assertion functions this line would just become something like:

```suggestion
assertMemoHasLimit(limit, memo)
```

Reply:
https://github.com/shapeshift/web/commit/8582fbd60c34e06b86104008523d3f7a69a26e54